crates/lib3h/src/transport_wss/mod.rs

```diff
@@ -265,7 +267,17 @@ impl<T: Read + Write + std::fmt::Debug> Transport for TransportWss<T> {
     /// this should be called frequently on the event loop
     /// looks for incoming messages or processes ping/pong/close events etc
     fn process(&mut self) -> TransportResult<(DidWork, Vec<TransportEvent>)> {
-        let did_work = self.priv_process_stream_sockets()?;
+        let mut did_work = false;
+
+        while !self.inbox.is_empty() {
+            let cmd = self.inbox.pop_front().expect("queue not empty");
```

Comment:
Is there a thread safe way to do this instead, like in a monad? I know we are single threaded but if we ever switch to multithreading this is the kinda bug which will bite us.

Reply:
good call :+1: gogo `while let`:

```rust
while let Some(ref cmd) = self.inbox.pop_front() {
    did_work = true;
    self.serve_TransportCommand(cmd)?;
}
```